### PR TITLE
Added support for TAAR profiles from Cloud BigTable

### PR DESCRIPTION
**Pull request by crankycoder**

### CODE REVIEW

File: taar_etl/taar_profile_bigtable.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+        self.BIGTABLE_TABLE_ID = bigtable_table_id
+
+        self.SAMPLE_RATE = sample_rate
+
+    def run_query(self, sql):
+        from google.cloud import bigquery
+
+        client = bigquery.Client()
+        job_config = bigquery.QueryJobConfig(use_query_cache=True)
+        query_job = client.query(sql, job_config=job_config)  # Make an API request.
```
Comment (haroldwoo) [inline taar_etl/taar_profile_bigtable.py:53]: do we need try/catch or retry here?

File: taar_etl/taar_profile_bigtable.py [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+            p | "Read" >> beam.io.ReadFromAvro(
+                gcs_avro_uri(self.GCS_BUCKET, self.NOW_ISODATE), use_fastavro=True
+            ) | "Create BigTable Rows" >> beam.Map(
+                create_bigtable_rows
+            ) | "Write Records to Cloud BigTable" >> WriteToBigTable(
+                project_id=self.GCP_PROJECT,
+                instance_id=self.BIGTABLE_INSTANCE_ID,
+                table_id=self.BIGTABLE_TABLE_ID,
+            )
+        print("Done!")
```
Comment (haroldwoo) [inline taar_etl/taar_profile_bigtable.py:124]: nit: done with what?

File: taar_etl/taar_profile_bigtable.py [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+    help="The BigQuery dataset ID that the user profile data will be held in prior to Avro export.",
+    default="cfr_etl",
+)
+@click.option(
+    "--bigquery-table-id",
+    help="The BigQuery table ID that the user profile data will be held in prior to Avro export.",
+    default="taar_profile",
+)
+@click.option(
+    "--gcs-bucket",
```
Comment (haroldwoo) [inline taar_etl/taar_profile_bigtable.py:288]: nit: naming more descriptive e.g. --avro-gcs-bucket

File: Makefile [hunk 4]
```diff
@@ -14,3 +17,8 @@ run_taar_amowhitelist:
 
 run_taar_update_whitelist:
 	docker run -t --rm -e AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID} -e AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY} app:build -m taar_etl.taar_update_whitelist --date 20190801
+
+
+push_gcr_io:
```
Comment (haroldwoo) [inline Makefile:22]: where will this be ran? locally for now? Ideally circleci will push this.

Comment (crankycoder) [reply Makefile:22]: Sorry - I should probably put some comments in that Makefile.  The intention of those `docker run` targets is just to run the containers under test.  I'll add some comments in.

File: taar_etl/taar_profile_bigtable.py (hunk 1 above)
Comment (crankycoder) [reply taar_etl/taar_profile_bigtable.py:53]: Do we have standard for number of retries in BigQuery?  I think I would prefer to see the job fail entirely if BigQuery fails for us and I can get logs that show me the failure.

File: taar_etl/taar_profile_bigtable.py (hunk 2 above)
Comment (crankycoder) [reply taar_etl/taar_profile_bigtable.py]: Thx.  Cleaned up that message to indicate BigTable load is completed.

**Comment by crankycoder** [comment]
@haroldwoo I'm happy to use either print or a logging class. I'm partial to *not* using logging for ETL jobs because of problems in logging that require a non-serializable mutex in the logging class that isn't fixed until python 3.8 https://bugs.python.org/issue30520 

Yes - the steps are all idempotent.

The imports in function are actually a requirement of Cloud Dataflow to prevent namespacing issues when functions are serialized and sent over the wire to other nodes in the cluster.

https://cloud.google.com/dataflow/docs/resources/faq#how_do_i_handle_nameerrors

File: taar_etl/taar_profile_bigtable.py (hunk 1 above)
Comment (crankycoder) [reply taar_etl/taar_profile_bigtable.py:53]: Sorry - I read this before I saw the last comment of the review.  Each step is idempotent so I don't think we need to deal with API request failures.

**Pull request closed without merging**